pennylane/_device.py

```diff
@@ -285,54 +285,91 @@ def supports_operation(self, operation):
         Returns:
             bool: ``True`` iff supplied operation is supported
         """
-        if isinstance(operation, type) and issubclass(operation, Operation):
-            return operation.__name__ in self.operations
-        if isinstance(operation, str):
-            return operation in self.operations
+        op_name = operation
 
-        raise ValueError("The given operation must either be a pennylane.Operation class or a string.")
+        if issubclass(operation.__class__, Operation):
+            op_name = operation.name
+        elif isinstance(operation, type):
+            op_name = operation.__name__
+        elif not isinstance(op_name, str):
+            raise ValueError("The given operation must either be a pennylane.Operation class or a string.")
```

Comment:
The old code was mine, so my bad that the docstring was wrong! The intention was to support both `supports_operation(qml.PauliX)` and `supports_operation("PauliX")`, so the accepted types should have been `(type,str)` no?

@antalszava why was this code changed in the first place?

Reply:
> the accepted types should have been (type,str) no?

I guess, but I wonder if there's a more specific syntax for "subclass of X"?

Reply:
The idea behind the change was to use the newly introduced `name` property of an `Operation` object such that it is apparent if the inverse of the operation or the operation itself was determined. Since, however, this `method` is now mainly used in the codebase with subclasses of `Operation`, the previous version is the correct one.

Modifying the docstring such that it contains `(type,str)`.